transaction/core/src/blockchain/block_version.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2018-2021 The MobileCoin Foundation
+
+use core::{fmt, hash::Hash, ops::Deref};
+use mc_crypto_digestible::Digestible;
+use serde::{Deserialize, Serialize};
+
+/// A block version number
+#[derive(
+    Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Digestible, Hash,
+)]
+pub struct BlockVersion(pub u32);
+
+impl From<u32> for BlockVersion {
+    fn from(src: u32) -> Self {
+        Self(src)
+    }
+}
+
+impl fmt::Display for BlockVersion {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
```

Comment:
Nit: Display -> Debug chains are not great for internationalization, so this should just print the version as digits and not worry about adding labels. We have debug for debug :-)

```suggestion
        write!(f, "{}", self.0)
```

Reply:
I just copied the code from `TokenId` here

Reply:
Lmk if you want that to change also

Reply:
Ideally, yeah, I've gone back and forth on this, and really I just want the display to show human readable contents, and let me fill in the details.

Reply:
Works for me